src/coord/src/coord.rs

```diff
@@ -4098,7 +4124,23 @@ impl Coordinator {
 
     async fn drop_indexes(&mut self, indexes: Vec<GlobalId>) {
         let mut trace_keys = Vec::new();
+        let mut compute_instance: Option<usize> = None;
+
         for id in indexes {
+            let inner = self
+                .catalog
+                .get_by_id(&id)
+                .index()
+                .expect("dropping index")
+                .compute_instance;
+            match compute_instance {
+                None => compute_instance = Some(inner),
+                Some(compute_instance) => assert_eq!(
+                    compute_instance, inner,
+                    "all dropped sinks must be on same instance"
+                ),
+            }
```

Comment:
Ah so by the time this is called the indexes will have been removed from the catalog. We need the caller of this method to supply the compute instance IDs. You can probably steal from my branch!

https://github.com/benesch/materialize/blob/c3eaa4d81725797852652acb1102f6d1c67c8d23/src/coord/src/coord.rs#L4194-L4232

https://github.com/benesch/materialize/blob/c3eaa4d81725797852652acb1102f6d1c67c8d23/src/coord/src/coord.rs#L4048-L4060

https://github.com/benesch/materialize/blob/c3eaa4d81725797852652acb1102f6d1c67c8d23/src/coord/src/coord.rs#L4106-L4111

Reply:
Derp, ty